### PR TITLE
feat: Support calling the OpenAI API directly

### PR DESCRIPTION
**Pull request by msnidal**
## Description

Support for going through the OpenAI API directly in addition to the existing Azure support (API and secrets). Also moves to the `model` notation as the `engine` param is deprecated and failed when hitting the OpenAI API directly.

Updates the README with more clarity on the order of operations

## Testing

TODO

## Additional context

I dont work at microsoft nor have an Azure account so I'm not sure how the engine vs. model distinction works on a custom Azure OpenAI endpoint 🤔

### CODE REVIEW

**Comment by msnidal** [comment]
@microsoft-github-policy-service agree

**Comment by dciborow** [comment]
@msnidal , looks great! Will get this sorted out and merged in.

**Comment by dciborow** [comment]
I realize now my isort has not been working. I am going to create a new PR that just takes care of all the import fixes.

File: src/gpt_review/context.py [hunk 1]
```diff
@@ -25,33 +23,44 @@ class Context:
...
-    openai.api_version = os.environ["OPENAI_API_VERSION"] = azure_config.get("azure_api_version", C.AZURE_API_VERSION)
+    if azure_config.get("azure_api_type"):
+        openai.api_type = os.environ["OPENAI_API_TYPE"] = azure_config.get("azure_api_type")
+    elif "OPENAI_API_TYPE" in os.environ:
+        openai.api_type = os.environ["OPENAI_API_TYPE"]
+
+    if azure_config.get("azure_api_version"):
+        openai.api_version = os.environ["OPENAI_API_VERSION"] = azure_config.get("azure_api_version")
+    elif "OPENAI_API_VERSION" in os.environ:
+        openai.api_version = os.environ["OPENAI_API_VERSION"]
```
Comment (dciborow) [inline src/gpt_review/context.py:51]:
```suggestion
    if azure_config.get("azure_api_type"):
        openai.api_type = os.environ["OPENAI_API_TYPE"] = azure_config.get("azure_api_type")
    elif "OPENAI_API_TYPE" in os.environ:
        openai.api_type = os.environ["OPENAI_API_TYPE"]
    elif "AZURE_OPENAI_API" in os.environ:
        openai.api_type = os.environ["OPENAI_API_TYPE"] = C.AZURE_API_TYPE

    if azure_config.get("azure_api_version"):
        openai.api_version = os.environ["OPENAI_API_VERSION"] = azure_config.get("azure_api_version")
    elif "OPENAI_API_VERSION" in os.environ:
        openai.api_version = os.environ["OPENAI_API_VERSION"]
    elif "AZURE_OPENAI_API" in os.environ:
        openai.api_type = os.environ["OPENAI_API_TYPE"] = C.AZURE_API_TYPE

```

Comment (dciborow) [reply src/gpt_review/context.py:51]: going to circle back around later and add a more global way to toggle if you want to use azure, but for now, we want to keep the file as optional and allow for either way.

File: src/gpt_review/context.py [hunk 2]
```diff
@@ -25,33 +23,49 @@ class Context:
...
+        openai.api_type = os.environ["OPENAI_API_TYPE"] = azure_config.get("azure_api_type")
+    elif "AZURE_OPENAI_API" in os.environ:
+        openai.api_type = os.environ["OPENAI_API_TYPE"] = C.AZURE_API_TYPE
+    elif "OPENAI_API_TYPE" in os.environ:
+        openai.api_type = os.environ["OPENAI_API_TYPE"]
+
+    if azure_config.get("azure_api_version"):
+        openai.api_version = os.environ["OPENAI_API_VERSION"] = azure_config.get("azure_api_version")
+    elif "AZURE_OPENAI_API" in os.environ:
+        openai.api_type = os.environ["OPENAI_API_TYPE"] = C.AZURE_API_TYPE
```
Comment (dciborow) [inline src/gpt_review/context.py:53]:
```suggestion
    elif "AZURE_OPENAI_API_VERSION" in os.environ:
        openai.api_version = os.environ["OPENAI_API_VERSION"] = C.AZURE_API_VERSION
```

**Comment by dciborow** [comment]
i was expecting the tests to run using my repos creds.... so that the tests would run properly.

**Review by dciborow — APPROVED** [review]

**Pull request merged**